ksqldb-engine/src/main/java/io/confluent/ksql/physical/pull/HARouting.java

```diff
@@ -200,7 +203,11 @@ private void executeRounds(
     while (processedPartitions < totalPartitions) {
       final Future<PartitionFetchResult> future = completionService.take();
       try {
-        final PartitionFetchResult fetchResult = future.get();
+        final long fetchTimeout = ThreadLocalRandom.current().nextLong(
+            KsqlConfig.KSQL_QUERY_PULL_FETCH_TIMEOUT_DEFAULT - 1,
```

Comment:
nit: we should make sure this isn't negative, right?

Reply:
Right, I was wondering if we have something like a verification check for the config variables. For instance, instead of checking for a negative value each time at this position one could require the timeout config value to be > 0 when parsing the config. I didn't spot anything like this though.